key_value_map.py

```diff
@@ -0,0 +1,105 @@
+from cartesian_coordinate import CartesianCoordinate
+from mapping_interface import MapInterface
+from mapping_interface import PointState
+
+
+class KeyValueMap(MapInterface):
+    """
+    @brief Class keyValueMap provides functionality to create a map using keys and values.
+    """
+
+    def __init__(self):
+        """
+        @brief keyValueMap constructor used to create a basic key value map.
+        """
+
+        self.map = {}
+        self.top_left_point = CartesianCoordinate(0, 0)
+        self.bot_right_point = CartesianCoordinate(0, 0)
+
+    def get_top_left_point(self):
+        """
+        @brief Method used to get the top_left_point of the key value map.
+        @return A CartesianCoordinate containing the top left corner point of the key value map.
+        """
+
+        return self.top_left_point
+
+    def get_bot_right_point(self):
+        """
+        @brief Method used to get the bot_right_point of the key value map.
+        @return A CartesianCoordinate containing the bottom right corner point of the key value map.
+        """
+
+        return self.bot_right_point
+
+    def get_height(self):
+        """
+        @brief Method used to get the height key value of the map.
+        @return A integer containing the height key value of the map.
+        """
+
+        return self.get_bot_right_point().y - self.get_top_left_point().y + 1
+
+    def get_width(self):
+        """
+        @brief Method used to get the width of the key value map.
+        @return A integer containing the width of the key value map.
+        """
+
+        return self.get_bot_right_point().x - self.get_top_left_point().x + 1
+
+    def get_map_points(self):
+        """
+        @brief Method used to get all the points stored in the map.
+        @return A list containing all the occupied points of the map.
+        """
+
+        return self.map.keys()
+
+    def is_occupied(self, coordinate : CartesianCoordinate) -> CartesianCoordinate:
+        """
+        @brief Checks whether a point is occupied in the key value map.
+        @return A boolean containing whether the point is occupied.
+        """
+
+        return coordinate in self.get_map_points()
+
+    def update_top_left_point(self, new_point):
```

Comment:
For consistency I suggest either always annotating your functions like you did with `is_occupied` or not doing it. Right now there isn't any way to know what `new_point` is, while if you annotated it, I'd be able to tell what you want. (is it a tuple? a list? a CatesianCoordinate?)